Merge end-to-end tests (#3537)
matrix-org/matrix-react-sdk: test/end-to-end-tests/README.md

> # Matrix React Web App End-to-End tests

Reviewer: ```suggestion
# Matrix React SDK End-to-End tests
```

> This repository contains tests for the matrix-react-sdk web app. The tests fire up a headless chrome and simulate user interaction (end-to-end). Note that end-to-end has little to do with the end-to-end encryption matrix supports, just that we test the full stack, going from user interaction to expected DOM in the browser.

Reviewer: ```suggestion
This directory contains tests for matrix-react-sdk. The tests fire up a headless Chrome and simulate user interaction (end-to-end). Note that end-to-end has little to do with the end-to-end encryption Matrix supports, just that we test the full stack, going from user interaction to expected DOM in the browser.
```

## Setup

Run `./install.sh`. This will:

>  - install synapse, fetches the master branch at the moment. If anything fails here, please refer to the synapse README to see if you're missing one of the prerequisites.

Reviewer: ```suggestion
 - install Synapse, fetches the master branch at the moment. If anything fails here, please refer to the Synapse README to see if you're missing one of the prerequisites.
```

>  - install riot, this fetches the master branch at the moment.

Reviewer: ```suggestion
 - install Riot, this fetches the master branch at the moment.
```

 - install dependencies (will download copy of chrome)

## Running the tests

Run tests with `./run.sh`.

### Debug tests locally.

> `./run.sh` will run the tests against the riot copy present in `riot/riot-web` served by a static python http server. You can symlink your `riot-web` develop copy here but that doesn't work well with webpack recompiling. You can run the test runner directly and specify parameters to get more insight into a failure or run the tests against your local webpack server.

Reviewer: ```suggestion
`./run.sh` will run the tests against the Riot copy present in `riot/riot-web` served by a static Python HTTP server. You can symlink your `riot-web` develop copy here but that doesn't work well with Webpack recompiling. You can run the test runner directly and specify parameters to get more insight into a failure or run the tests against your local Webpack server.
```

```
./synapse/stop.sh && \
./synapse/start.sh && \
node start.js <parameters>
```

> It's important to always stop and start synapse before each run of the tests to clear the in-memory sqlite database it uses, as the tests assume a blank slate.

Reviewer: ```suggestion
It's important to always stop and start Synapse before each run of the tests to clear the in-memory SQLite database it uses, as the tests assume a blank slate.
```

start.js accepts these parameters (and more, see `node start.js --help`) that can help running the tests locally:

>  - `--riot-url <url>` don't use the riot copy and static server provided by the tests, but use a running server like the webpack watch server to run the tests against. Make sure to have the following local config:

Reviewer: ```suggestion
 - `--riot-url <url>` don't use the Riot copy and static server provided by the tests, but use a running server like the Webpack watch server to run the tests against. Make sure to have the following local config:
```
